### PR TITLE
IPMPROG-4695 - csv export file labels fixed

### PR DESCRIPTION
**Pull request by sinanyvs**

### CODE REVIEW

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

**Review by ericclappier-eaton — DISMISSED** [review]

File: ecpp/average.ecpp [hunk 1]
```diff
@@ -308,12 +309,30 @@ bool database_ready;
...
+                        // first row: "Time (Sec)" label + timestamps receipt (index 0)
+                        csv_data.push_back (std::vector <std::string> {"Time (UTC)"});
                     }
 
                     // add a new row: type_rep label + datas
-                    csv_data.push_back (std::vector <std::string> {type_rep});
+
+                    std::string correctedType(type_rep);
+
+                    if(checked_source == "realpower.default")
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: a detail, usually we add a space separator after the if else while for....  ;)
if (...

File: ecpp/average.ecpp [hunk 2]
```diff
@@ -322,7 +341,12 @@ bool database_ready;
                         char *value = zmsg_popstr (recv_msg);
 
                         if (!got_timestamps) {
-                            csv_data [0].push_back (timestamp);
+                            time_t rawtime = atoi(timestamp);
+                            struct tm timeinfo;
+                            localtime_r(&rawtime, &timeinfo);
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp:362]: it seems that localtime_r() (POSIX) is not thread-safe... 
you should use strftime() instead.

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

File: ecpp/average.ecpp (hunk 2 above)
Comment (sinanyvs) [reply ecpp/average.ecpp:362]: They don't seem to be alternatives. localtime converts seconds from epoch(time_t) to tm struct. even if I use strftime I need to do conversion.

**Review by aquette — DISMISSED** [review]
Lgtm, thx

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

File: ecpp/average.ecpp [hunk 3]
```diff
@@ -72,6 +73,36 @@ bool relative_to_unixtime (const std::string& relative, int64_t now, int64_t& un
     return false;
 }
 
+std::string getCorrectedType(const std::string & type,const std::string & src) {
+
+    std::string result(type);
+
+    static const std::map<std::string, std::string> typeMap = {
+     {"arithmetic_mean", "Usage mean"},
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: tab 4 spaces....

File: ecpp/average.ecpp [hunk 4]
```diff
@@ -72,6 +73,36 @@ bool relative_to_unixtime (const std::string& relative, int64_t now, int64_t& un
...
+
+    static const std::map<std::string, std::string> unitMap = {
+     {"realpower.default", "(Watt)"},
+     {"current.output.L1", "(A)"},
+     {"voltage.output.L1-N", "(V)"},
+     {"power.default", "(VA)"}
+    };
+
+    if (unitMap.count(src))
+        {
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: tabs and open brace...

File: ecpp/average.ecpp [hunk 5]
```diff
@@ -352,12 +391,28 @@ bool database_ready;
         //escape special characters
         element_ename = std::regex_replace(element_ename, std::regex("( |\t)"), "_");
 
+        static const std::map<std::string, std::string> srcMap = {
+                                                                   {"realpower.default", "power-usage"},
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: for readability... trim left this map definition

File: ecpp/average.ecpp [hunk 6]
```diff
@@ -72,6 +73,36 @@ bool relative_to_unixtime (const std::string& relative, int64_t now, int64_t& un
...
 
+std::string getCorrectedType(const std::string & type,const std::string & src) {
+
+    std::string result(type);
+
+    static const std::map<std::string, std::string> typeMap = {
+     {"arithmetic_mean", "Usage mean"},
+     {"min", "Usage min"},
+     {"max", "Usage max"},
+     {"consumption", "Consumption(Watt-second)"}
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: space after Consumption

File: ecpp/average.ecpp [hunk 7]
```diff
@@ -72,6 +73,36 @@ bool relative_to_unixtime (const std::string& relative, int64_t now, int64_t& un
...
+     {"power.default", "(VA)"}
+    };
+
+    if (unitMap.count(src))
+        {
+            if (typeMap.count(result)) {
+                result = typeMap.at(result);
+
+            if(type != "consumption")
+                result += unitMap.at(src);
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp:102]: missed a space between type and unit

File: ecpp/average.ecpp [hunk 8]
```diff
@@ -308,12 +339,15 @@ bool database_ready;
...
+                        // first row: "Time (Sec)" label + timestamps receipt (index 0)
+                        csv_data.push_back (std::vector <std::string> {"Time (UTC)"});
                     }
 
                     // add a new row: type_rep label + datas
-                    csv_data.push_back (std::vector <std::string> {type_rep});
+
+                    std::string correctedType = getCorrectedType(type_rep, checked_source);
+
+                    csv_data.push_back (std::vector <std::string> {correctedType.c_str()});
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/average.ecpp]: csv_data.push_back( { correctedType } );

File: ecpp/average.ecpp (hunk 3 above)
Comment (sinanyvs) [reply ecpp/average.ecpp]: done

File: ecpp/average.ecpp (hunk 4 above)
Comment (sinanyvs) [reply ecpp/average.ecpp]: done

File: ecpp/average.ecpp (hunk 5 above)
Comment (sinanyvs) [reply ecpp/average.ecpp]: done

File: ecpp/average.ecpp (hunk 6 above)
Comment (sinanyvs) [reply ecpp/average.ecpp]: done

File: ecpp/average.ecpp (hunk 7 above)
Comment (sinanyvs) [reply ecpp/average.ecpp:102]: done

File: ecpp/average.ecpp (hunk 8 above)
Comment (sinanyvs) [reply ecpp/average.ecpp]: done

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

**Pull request merged**